### PR TITLE
docs: picker

### LINKED ISSUE #418: pydocs for picker
https://react-spectrum.adobe.com/react-spectrum/Picker.html

### PR DESCRIPTION
**Pull request by ethanalvizo**
Closes #418

### CODE REVIEW

**Comment by ethanalvizo** [comment]
Accidentally based this branch off of my form pydocs branch. Should merge https://github.com/deephaven/deephaven-plugins/pull/602 first then this one.

File: plugins/ui/src/deephaven/ui/components/picker.py [hunk 1]
```diff
@@ -6,9 +6,25 @@
 from .basic import component_element
 from .section import SectionElement, Item
 from .item_table_source import ItemTableSource
-from ..elements import BaseElement
+from ..elements import BaseElement, Element
 from .._internal.utils import create_props, unpack_item_table_source
 from ..types import Key
+from .types import (
+    # Layout
```
Comment (mofojed) [inline plugins/ui/src/deephaven/ui/components/picker.py:13]:
```suggestion
```

File: plugins/ui/src/deephaven/ui/components/picker.py [hunk 2]
```diff
@@ -27,7 +43,83 @@ def picker(
...
+    direction: MenuDirection = "bottom",
+    should_flip: bool = True,
+    menu_width: DimensionValue | None = None,
+    auto_focus: bool | None = None,
+    auto_complete: str | None = None,
+    name: str | None = None,
+    is_open: bool | None = None,
+    default_open: bool | None = None,
+    # items
+    # disabled_keys
```
Comment (mofojed) [inline plugins/ui/src/deephaven/ui/components/picker.py:57]:
```suggestion
```

File: plugins/ui/src/deephaven/ui/components/picker.py [hunk 3]
```diff
@@ -27,7 +43,83 @@ def picker(
...
+    is_disabled: bool | None = None,
+    is_required: bool | None = None,
+    is_invalid: bool | None = None,
+    validation_behavior: ValidationBehavior | None = None,
+    # validate # omitted because it has synchronous return
```
Comment (mofojed) [inline plugins/ui/src/deephaven/ui/components/picker.py:62]:
```suggestion
```

File: plugins/ui/src/deephaven/ui/components/picker.py [hunk 4]
```diff
@@ -27,7 +43,83 @@ def picker(
...
+    auto_complete: str | None = None,
+    name: str | None = None,
+    is_open: bool | None = None,
+    default_open: bool | None = None,
+    # items
+    # disabled_keys
+    is_disabled: bool | None = None,
+    is_required: bool | None = None,
+    is_invalid: bool | None = None,
+    validation_behavior: ValidationBehavior | None = None,
```
Comment (mofojed) [inline plugins/ui/src/deephaven/ui/components/picker.py:61]:
```suggestion
    validation_behavior: ValidationBehavior | None = "aria",
```

File: plugins/ui/src/deephaven/ui/components/picker.py [hunk 5]
```diff
@@ -27,7 +43,83 @@ def picker(
...
+    label_alignment: Alignment = "start",
+    necessity_indicator: NecessityIndicator = "icon",
+    contextual_help: Element | None = None,
+    on_open_change: Callable[[bool], None] | None = None,
+    on_focus: FocusEventCallable | None = None,
+    on_blur: FocusEventCallable | None = None,
+    on_focus_change: Callable[[bool], None] | None = None,
+    on_key_down: KeyboardEventCallable | None = None,
+    on_key_up: KeyboardEventCallable | None = None,
+    # on_load_more # omitted because it has synchronous return
```
Comment (mofojed) [inline plugins/ui/src/deephaven/ui/components/picker.py:78]: This is incorrect, `on_load_more` can actually be an async return: https://react-spectrum.adobe.com/react-spectrum/Picker.html#asynchronous-loading

That being said, we're not supporting `items`, and `on_load_more` works with `items`, so just drop this.
```suggestion
```

**Review by mofojed — CHANGES_REQUESTED** [review]

File: plugins/ui/src/deephaven/ui/components/picker.py (hunk 4 above)
Comment (ethanalvizo) [reply plugins/ui/src/deephaven/ui/components/picker.py:58]: Yes will create this PR when I get back. But all the other comments are addressed

**Review by mofojed — APPROVED** [review]

File: plugins/ui/src/deephaven/ui/components/picker.py (hunk 4 above)
Comment (ethanalvizo) [reply plugins/ui/src/deephaven/ui/components/picker.py:58]: https://github.com/adobe/react-spectrum/pull/6701

**Pull request merged**